### PR TITLE
Fix Battler Z ordering

### PR DESCRIPTION
**Pull request by fmatthew5876**
Fix: #1791

### CODE REVIEW

**Comment by Ghabry** [comment]
There is already a SetZ call here: https://github.com/EasyRPG/Player/blob/51c2598d0d7139532519120727d547f7368ad4f4/src/sprite_battler.cpp#L310

Can Z change during the battle, in the old code it is only set once on creation.

File: src/game_battler.cpp [hunk 1]
```diff
@@ -785,6 +785,21 @@ int Game_Battler::GetDisplayY() const {
...
 }
 
+int Game_Battler::GetDisplayZ() const {
+	static_assert(Game_Battler::Type_Ally < Game_Battler::Type_Enemy, "Game_Battler enums re-ordered! Fix Z order logic here!");
+
+	constexpr int id_limit = 128;
+
+	int z = GetType();
+	z *= SCREEN_TARGET_HEIGHT * 2;
+	z += GetDisplayY();
```
Comment (carstene1ns) [inline src/game_battler.cpp]: Might be better to use `GetBattleY()` here, since it is unaffected from flying.

**Comment by fmatthew5876** [comment]
The use of `GetId()` is wrong. For enemies it returns the enemy id, not the troop id.

For the algorithm, enemies use troop id, but actors always use actor id, even if the part is ordered differently.

**Comment by fmatthew5876** [comment]
Tested this more thoroughly and pushed new fixes

* For enemies, Y position considered is actually the bottom edge of the sprite, not the centered y position
* Flying offset is ignored
* Actors used actor id, regardless of party order
* Enemies use troop id

For actors, I ignored graphic size, since all actors have same size graphics and the graphic isn't properly initialized at the time of the call.

**Pull request merged**